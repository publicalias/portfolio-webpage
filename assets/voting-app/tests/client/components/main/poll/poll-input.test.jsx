"use strict";

//local imports

const PollInput = require("../../../../../scripts/client/components/main/poll/poll-input");

const { initTestPoll, testReload, testWrapper } = require("../../../test-helpers");

//global imports

const { initTestEvent, initTestSnapshot, testSubmit } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(PollInput);

const testForm = initTestPoll(testShallow, "form");
const testView = initTestPoll(testShallow, "view");

const initTestChange = (render) => (type) => {

  const testChange = initTestEvent(render, "change", { target: { value: "Option A" } });

  return testChange(".qa-change-input", [], [type, ["Option A"]]);

};

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(PollInput));

//poll input

describe("PollInput (form)", () => {

  const testSnapshot = initTestSnapshot(testForm);

  const testChange = initTestChange(testForm);
  const testClick = initTestEvent(testForm, "click");

  it("should match snapshot", () => testSnapshot());

  it("should call handleChange on change", () => testChange("formSetAdd"));

  it("should call handleSubmit on click", () => {

    const dataList = [{ form: { add: "Option A" } }];

    return testClick(".qa-submit-input", dataList, ["formAddOption", ["Option A", []]]);

  });

});

describe("PollInput (view)", () => {

  const testSnapshot = initTestSnapshot(testView);

  const testChange = initTestChange(testView);

  it("should match snapshot", () => testSnapshot());

  it("should call handleChange on change", () => testChange("viewSetAdd"));

  testSubmit("click", "handleSubmit", (res) => testReload(
    testView,
    [
      { view: { add: "Option A" } },
      { id: "id-a" },
      null,
      { actions: { pollAddOption: jest.fn(() => res) } }
    ],
    ".qa-submit-input",
    "id-a",
    ["pollAddOption", ["id-a", "Option A"]],
    ["viewSetAdd", res && !res.errors ? [""] : undefined]
  ));

});
