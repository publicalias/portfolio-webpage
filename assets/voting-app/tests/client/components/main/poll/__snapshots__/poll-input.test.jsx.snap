// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollInput (form) should match snapshot 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option (Up to 20)"
    value=""
  />
  <button
    className="c-poll-options__submit qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`PollInput (view) should match snapshot 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option (Up to 20)"
    value=""
  />
  <button
    className="c-poll-options__submit qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
